packages/trpc/server/createContext.ts

```diff
@@ -62,7 +62,7 @@ export async function createContextInner(opts: CreateInnerContextOptions) {
  * @link https://trpc.io/docs/context
  */
 export const createContext = async ({ req, res }: CreateContextOptions, sessionGetter?: GetSessionFn) => {
-  const locale = getLocaleFromHeaders(req);
+  const locale = await getLocaleFromRequest(req);
```

Comment:
Is now async due to session checks. It should not add too much overhead since sessions are cached.